pkg/pipelinerun/tracker.go

```diff
@@ -107,8 +106,36 @@ func (t *Tracker) Monitor(allowed []string) <-chan []taskrunpkg.Run {
 
 	informer.AddEventHandler(
 		cache.ResourceEventHandlerFuncs{
-			AddFunc:    eventHandler,
-			UpdateFunc: func(_, newObj interface{}) { eventHandler(newObj) },
+			AddFunc: func(obj interface{}) {
+				mu.Lock()
+				defer mu.Unlock()
+				select {
+				case <-stopC:
+					return
+				default:
+					eventHandler(obj)
+				}
+			},
+			UpdateFunc: func(_, newObj interface{}) {
+				mu.Lock()
+				defer mu.Unlock()
+				select {
+				case <-stopC:
+					return
+				default:
+					eventHandler(newObj)
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
```

Comment:
can you please add comments so that it can be clear to the person who visits this code next time